### DISCUSSION TITLE
PinePhone with RNS + NomadNet

### DISCUSSION THREAD

**Discussion by The-One-Law** [markqvist/Reticulum#229] [Nomad Network]
Just in case anyone needs to know,
 documenting a bit of about 
### Installing and running NomadNet on PinePhone.

Running **Manjaro Phosh** (on what was originally sold as Mobian)
 for 2 years now.
 (after testing and 'noping' notable alternatives).
see:
https://wiki.pine64.org/wiki/PinePhone_Software_Releases

![5bda_P_20230215_005136_1920p](https://user-images.githubusercontent.com/49971941/218899996-f0c79a87-ef28-4f43-8f5a-270042d73c3f.jpg)

There may be better improvements elsewhere since I locked onto Manjaro,
 but it depends _heavily on your patience_ and preferences.
None of it is for the faint of heart - the learning curve can be brutal.

---    
Some of my bash history which (possibly, or not !) may be crucial:

in a console, (it is a gnome console)
 select the "terminal mode" onscreen keyboard, unless you use a hardware keyboard.
 
sudo pacman -S pinephone-devtools
 ( you may not need this, but it may have enabled unknown bits, so just noting...)

sudo pacman -S python-pip
sudo pacman -S gcc  (gcc is necessary)

you may want to install  mc  if you do not have it

pip3 install rns
pip3 install nomadnet

then simply run:
 nomadnet

If it runs okay, quit - then add a [testnet interface ](https://github.com/markqvist/Reticulum/blob/master/docs/source/gettingstartedfast.rst#connect-to-the-public-testnet)
to the end of your
 ~/.reticulum/config file
so you can communicate with the rest of us.
You may need to reboot to activate the testnet interface. 

See you there, occasionally.

Best go into settings and set your display to 1.5 (2.0 really does not work for Nomadnet) 
(if you try 1.0 you will need a stylus _to do anything_...)
nomadnet uses urwid - still seeking a reasonable method to set canvas size - urwid is '_full auto_'.

I do not know about using nomadnet over the telephony modem:
 - I avoid using sim cards (they do work) because I suspect
 that networks will blacklist my imei before some unforeseen urgency when I may actually need it.
And some telcos will blacklist your gear for random whims. (...been there, done that)
I do use the modemswitch "on" for GPS which works (sporadically) without a SIM Card (or Google).
 
**DO understand that changing between portrait and landscape mode can kill your nomadnet session.
This seems to be something that Pinephone does on its own, but urwid is also a possibility.**
(The urwid usage has a delay for flipping screen sizes and that may be where it breaks.) 

After you kill a few sessions you should reboot to clear a lot of lost threads.
Just a habit that will keep you from wasting hours chasing a tail.
It is a "PinePhone rubbish disposal method".

Control -e is what you use to view  node info in a Conversations window.
It seems to also (randomly) kill the session,
 but how this happens I have not been able to find.
Staring at "unhandled_input" just makes my eyes water,
 so I guess that waits another day.

**Comment by Erethon** [comment]
> Control -e is what you use to view node info in a Conversations window. It seems to also (randomly) kill the session, but how this happens I have not been able to find. Staring at "unhandled_input" just makes my eyes water, so I guess that waits another day.

This sounds like it could be related to https://github.com/markqvist/NomadNet/issues/22 which is now fixed, **but** there's no nomadnet release containing the fix yet. You can try installing Nomadnet through the git repo to see if this fixes it for you.

**Comment by githubques** [comment]
Thanks for figuring this out. Have you thought about posting this over on the Pine64 Manjaro Sub-Forum to get more people using this? https://forum.pine64.org/forumdisplay.php?fid=130

**Reply by The-One-Law** [reply]
Silly me,
 I had avoided posting there because I had little to contribute to the woes that are PinePhone.
I have been a consumer there with nothing constructive to add to the conversation there.
This little bit here is a whole different exercise.
I shall proceed to venture forth there.
Thank you for the reminder.

https://forum.pine64.org/showthread.php?tid=17928

**Comment by The-One-Law** [comment]
Just caught up on this (I live a non-linear existence).

 installing gcc is necessary to fix an issue (python problem, I guess)  with netifaces.
(see: https://forum.pine64.org/showthread.php?tid=17928&pid=116075#pid116075 )

Just pointing this out - the original instructions still stand.

cheers
